server/common/annotations/hosted_tiledb.py

```diff
@@ -93,20 +107,20 @@ def convert_to_pandas_df(self, tileDBArray):
         return new_df
 
     def write_labels(self, df, data_adaptor):
-
         user_id = current_app.auth.get_user_id()
+        user_name = current_app.auth.get_user_name().replace(' ', '')
         timestamp = time.time()
-        dataset_name = data_adaptor.get_location()
-        dataset_id = self.db.get_or_create_dataset(dataset_name)
+        dataset_location = data_adaptor.get_location()
+        dataset_id = self.db.get_or_create_dataset(dataset_location)
+        dataset_name = data_adaptor.get_title()
         user_id = self.db.get_or_create_user(user_id)
 
-        uri = f"{self.directory_path}-{dataset_name}-{user_id}-{timestamp}"
+        uri = f"{self.directory_path}{dataset_name}/{user_name}/{timestamp}"
```

Comment:
I had a general question about naming of annotation files; I understand that it might be easier to debug if you have filenames that we can quickly take a look at but to me this is not a good enough reason to add a bunch of rules in the business logic tier of the application that should be at the data access layer. We are starting to blur the lines which inevitably will lead to faulty code. For example, `time.time()` is almost guaranteed to cause an issue at some point because the [clocks on different servers aren’t always set exactly the same](https://docs.python.org/2/library/time.html#time.time). I’d advocate strongly to revise such that the database keep track of the user id, dataset id, etc. and use `uuid4()` to generate filenames. Or at the very least, tag on a `uuid4()` id to the filename.

Reply:
So the time on the filename is just used as a differentiator between files for a user/dataset. The time the annotation is saved to the db (created_at) is the time we sort by. The database is the place we are keeping track of user id/dataset id etc. When retrieving annotations we read from the db to get the uri. 
I can append a uuid4() to the end of the filename to alleviate any concerns about collisions.

Reply:
discussed further over slack